src/core/ipns/resolver.js

```
@@ -102,30 +83,35 @@ class IpnsResolver {
       }
 
       // continue recursively until depth equals 0
-      this.resolver(nameSegments[2], depth - 1, peerId, resolverFn, callback)
+      this.resolver(nameSegments[2], depth - 1, callback)
     })
   }
 
-  // resolve ipns entries locally using the datastore
-  _resolveLocal (name, peerId, callback) {
-    const { ipnsKey } = ipns.getIdKeys(fromB58String(name))
+  // resolve ipns entries from the provided routing
+  _resolveName (name, callback) {
+    const peerId = PeerId.createFromB58String(name)
+    const { routingKey } = ipns.getIdKeys(peerId.toBytes())
 
-    this._repo.datastore.get(ipnsKey, (err, dsVal) => {
-      if (err) {
-        const errMsg = `local record requested was not found for ${name} (${ipnsKey})`
+    // TODO DHT - get public key from routing?
+    // https://github.com/ipfs/go-ipfs/blob/master/namesys/routing.go#L70
+    // https://github.com/libp2p/go-libp2p-routing/blob/master/routing.go#L99
+
+    this._routing.get(routingKey.toBuffer(), (err, res) => {
+      if (err || !res) {
```

Please split into two if statements so we don't mask errors as not found.